### PR TITLE
Arnej/add serialized fast value attribute

### PR DESCRIPTION
**Pull request by arnej27959**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@geirst please review

### CODE REVIEW

File: searchlib/src/vespa/searchlib/tensor/serialized_fast_value_attribute.cpp [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+                 from_store.num_subspaces)
+    {
+        assert(_type.cell_type() == _cells.type);
+        std::vector<vespalib::stringref> address(from_type.num_mapped_dimensions);
+        auto block_stream = ValueBlockStream(from_type, from_store);
+        size_t ss = 0;
+        while (auto block = block_stream.next_block()) {
+            size_t idx = my_index.map.add_mapping(block.address);
+            if (idx != ss) {
+                LOG(error, "add_mapping returned idx=%zu for subspace %zu", idx, ss);
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/tensor/serialized_fast_value_attribute.cpp:86]: Consider adding labels to error message.

File: searchlib/src/vespa/searchlib/tensor/streamed_value_store.cpp [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+        const char *aligned_ptr = fix_alignment(source.peek(), alignment);
+        size_t adjustment = aligned_ptr - source.peek();
+        source.adjustReadPos(adjustment);
+    }
+    retval.cells_ref = TypedCells(source.peek(), _data_from_type.cell_type, num_cells);
+    source.adjustReadPos(CellTypeUtils::mem_size(_data_from_type.cell_type, num_cells));
+    retval.num_subspaces = source.readValue<uint32_t>();
+    retval.labels_buffer = vespalib::ConstArrayRef<char>(source.peek(), source.size());
+ 
+    if (retval.num_subspaces * _data_from_type.dense_subspace_size == num_cells) {
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/tensor/streamed_value_store.cpp:145]: Change to an assert.

File: searchlib/src/vespa/searchlib/tensor/streamed_value_store.cpp [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+        retval.valid = true;
+        return retval;
+    }
+    LOG(warning, "inconsistent stored tensor data: "
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/tensor/streamed_value_store.cpp:149]: We can remove the log message with the assert above.

**Review by geirst — APPROVED** [review]
👍

**Comment by geirst** [comment]
@toregge FYI. This will replace SerializedTensorAttribute and SerializedTensorStore.

**Comment by arnej27959** [comment]
@geirst please take another look after follow-up fixes (on Monday)
@havardpe please review also (on Monday)

**Comment by arnej27959** [comment]
Note: we could save memory in the old attribute by using the new store also, but I chose not to include that change.

**Comment by geirst** [comment]
👍

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**